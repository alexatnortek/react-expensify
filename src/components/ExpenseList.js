import React from "react";
import { connect } from 'react-redux';
import ExpenseListItem from "./ExpenseListItem";
import selectExpenses from '../selectors/expenses'

export const ExpenseList = (props) => (
  <div>
    {
      props.expenses.length === 0 
      ?  
      (
        <p>No expenses</p>
      ) 
      : 
      (
        props.expenses.map(expense => {
          return <ExpenseListItem key={expense.id} {...expense} />
        })
      )
    }
  </div>
)

const mapStateToProps = state => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
    // expenses: state.expenses,
    // filters: state.filters
  }
}
export default connect(mapStateToProps)(ExpenseList);

// above is the same as below!
// const ConnectedExpenseList = connect(state => {
// export default connect(state => {  
//   return {
//     expenses: state.expenses
//   }
// })(ExpenseList);

// export default ConnectedExpenseList;

    // {props.expenses.map((expense, index) => (
    //   <div>
    //     <ExpenseListItem
    //     description={expense.description}
    //     amount={expense.amount}
    //     createdAt={expense.createdAt}
    //     />
    //   </div>
    // ))
    // }