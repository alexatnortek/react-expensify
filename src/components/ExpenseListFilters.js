import React from "react";
import { connect } from "react-redux";
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from "../actions/filters";
import { DateRangePicker } from "react-dates";
import uuid from 'uuid';

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  }
  onDatesChange = ({ startDate, endDate}) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = (calendarFocused) => {
    this.setState((() => ({ calendarFocused })))
  };
  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
    // console.log(e.target.value);
  };
  onSortChange = (e) => {
    e.target.value === 'amount' ? this.props.sortByAmount() : this.props.sortByDate();
  };
  render() {
    return <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
              <input type="text" 
              className="text-input"
              placeholder="Search expenses"
              value={this.props.filters.text} onChange={this.onTextChange} />
          </div>
          <div className="input-group__item">
            <select className="select" value={this.props.filters.sortBy} onChange={this.onSortChange}>
              <option value="amount">Amount</option>
              <option value="date">Date</option>
            </select>
          </div>
          <div className="input-group__item">
            <DateRangePicker startDate={this.props.filters.startDate} // startDateId={uuid()}
              startDateId={"startDate"} endDate={this.props.filters.endDate} // endDateId={uuid()}
              endDateId={"endDate"} onDatesChange={this.onDatesChange} focusedInput={this.state.calendarFocused} showClearDates={true} onFocusChange={this.onFocusChange} numberOfMonths={1} isOutsideRange={() => false} />
          </div>
        </div>
      </div>;
  }
}

// const mapStateToProps = state => {
//   return {
//     filters: state.filters
//   }
// };

//same as above
const mapStateToProps = state => ({
  filters: state.filters
});

const mapDispatchToProps = dispatch => ({
  setTextFilter: text => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setStartDate: startDate => dispatch(setStartDate(startDate)),
  setEndDate: endDate => dispatch(setEndDate(endDate))
})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);