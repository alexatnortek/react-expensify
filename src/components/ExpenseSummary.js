import React from 'react';
import { Link } from "react-router-dom";
import {connect} from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import getTotalExpenses from '../selectors/expenses-total';


export const ExpenseSummary = ( { expenseCount, totalExpenses} ) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
  const formattedTotalExpenses = numeral(totalExpenses / 100).format('$0,0.00');
  return <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">
          Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>
            {formattedTotalExpenses}
          </span>
        </h1>
        <div className="page-header__actions">
          <Link className="button-layout" to="/create">Add Expense</Link>
        </div>
      </div>
    </div>;
}


// export const ExpenseSummary = props => {
//   console.log(props);
//   return (
//     <div>
//       <p>Here is new things</p>
//         <p>
//           Viewing ${props.expenseCount} expenses totalling ${
//             props.totalExpenses
//           }
//         </p>
//     </div>
//   );
// };


const mapStateToProps = state => {

  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    totalExpenses: getTotalExpenses(visibleExpenses)
  }
}

export default connect(mapStateToProps)(ExpenseSummary);