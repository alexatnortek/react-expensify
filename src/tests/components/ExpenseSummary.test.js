import React from 'react';
import { shallow } from "enzyme";
import { ExpenseSummary } from "../../components/ExpenseSummary";


test('should correctly render expenseSummary with one expense', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={1} totalExpenses={1001} />);
  expect(wrapper).toMatchSnapshot();
});

test("should correctly render expenseSummary with multiple expenses", () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={2} totalExpenses={2002} />);
  expect(wrapper).toMatchSnapshot();

});