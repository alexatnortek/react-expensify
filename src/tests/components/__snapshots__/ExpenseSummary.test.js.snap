// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expenseSummary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        2
      </span>
       
      expenses
       totalling 
      <span>
        $20.02
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button-layout"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render expenseSummary with one expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $10.01
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button-layout"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
